### ISSUE TITLE
Nesting and frames support

### ISSUE THREAD

**Issue by OR13** [mattrglobal/jsonld-signatures-bbs#91]
```
{
  "@context": {"a": "https://example.com/a", "a1": "https://example.com/a1", "b": "https://example.com/b", "c": "https://example.com/c"},
  "a":{
    "b": {
      "c": "hello"
    }
  },
  "a1": "hello2"
}
```

```
{
  "@context": {"a": "https://example.com/a", "a1": "https://example.com/a1", "b": "https://example.com/b", "c": "https://example.com/c"},
 "@explicit": true,
  "a": {
   "@explicit": true,
    "b":{}
   }
}
```

trying to figure out how to get deriveProof on nest claims, for example:

```
{
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://www.w3.org/2018/credentials/examples/v1",
      "https://w3c-ccg.github.io/ldp-bbs2020/context/v1"
    ],
    "id": "http://example.gov/credentials/3732",
    "type": [
      "VerifiableCredential",
      "UniversityDegreeCredential"
    ],
    "issuer": {
      "id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s"
    },
    "issuanceDate": "2020-03-10T04:24:12.164Z",
    "credentialSubject": {
      "id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM",
      "degree": {
        "type": "BachelorDegree",
        "name": "Bachelor of Science and Arts"
      }
    },
    "proof": {
      "type": "BbsBlsSignature2020",
      "created": "2020-12-06T16:10:24Z",
      "proofPurpose": "assertionMethod",
      "proofValue": "sKNVGJssWdZcvmCqvcfgz7s64Ibj0ZfqSmZIwgy+yIhCUOlWflsBKCfVNGAPfd8yW/pryt87c7ho/PAJofdFZZB97gpQkpisI/+Rcs2xLtVRrpTiu4YU3sXEFbgPjA3GY/fZOpdFTABVutDvckchLw==",
      "verificationMethod": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM"
    }
  }
```

with 

```
   const deriveProofFrame = {
      "@context": [
        "https://www.w3.org/2018/credentials/v1",
        "https://www.w3.org/2018/credentials/examples/v1",
        "https://w3c-ccg.github.io/ldp-bbs2020/context/v1",
      {
        "degree": "https://example.com/degree"
      }
      ],
      "type": ["VerifiableCredential", "UniversityDegreeCredential"],
      "@explicit": true,
      "credentialSubject": {
        "@explicit": true,
        degree: {
          name: {}
        }
      }
    }

```

**Comment by kdenhartog** [comment]
Noting I saw this, but need to play around with things a bit before I can come back with an answer. I think this is mainly going to come down to upgrading how we use the frame API in json-ld or updating json-ld.js

**Comment by kdenhartog** [comment]
Did some digging on this so far, but stopping for the day. This is the value of compactInputProofDocument that's being [passed into jsonld.frame on line 138](https://github.com/mattrglobal/jsonld-signatures-bbs/blob/2673a0a077c232ca5be61b93339be547e5341635/src/BbsBlsSignatureProof2020.ts#L138)

This seems a bit odd to me so I need to do some more digging into if the fromRDF value is getting bad inputs which is causing the problem or if the framing is having some issues with it's inputs.

One thing that seems odd about this is some of the nodes are properly labelled `urn:bnid:_:c14n0` where as other's aren't being converted properly such as `"@id": "_:c14n0"`. Could be our blank node solution is biting us now and we need to re-evaulate it as well.

```
[{
        "@id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM",
        "https://example.org/examples#degree": [{
            "@id": "_:c14n0"
        }]
    },
    {
        "@id": "https://example.gov/credentials/3732",
        "@type": [
            "https://example.org/examples#UniversityDegreeCredential",
            "https://www.w3.org/2018/credentials#VerifiableCredential"
        ],
        "https://www.w3.org/2018/credentials#credentialSubject": [{
            "@id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM"
        }],
        "https://www.w3.org/2018/credentials#issuanceDate": [{
            "@value": "2020-03-10T04:24:12.164Z",
            "@type": "http://www.w3.org/2001/XMLSchema#dateTime"
        }]
    },
    {
        "@id": "urn:bnid:_:c14n0",
        "http://schema.org/name": [{
            "@value": "Bachelor of Science and Arts",
            "@type": "http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML"
        }],
        "@type": [
            "https://example.org/examples#BachelorDegree"
        ]
    }
]
```

More to come on this next week!

**Comment by OR13** [comment]
ping @msporny @dlongley if you have thoughts

**Comment by kdenhartog** [comment]
This will be fixed in #96

Thanks for raising this @OR13
